Projects/ChangeTier/Sources/Views/EditTier.swift

```diff
@@ -0,0 +1,82 @@
+import SwiftUI
+import hCore
+import hCoreUI
+
+struct EditTier: View {
+    @State var selectedTier: String?
+    var vm: SelectTierViewModel
+    @EnvironmentObject var selectTierNavigationVm: ChangeTierNavigationViewModel
+
+    init(
+        vm: SelectTierViewModel
+    ) {
+        self.vm = vm
+    }
+
+    var body: some View {
+        hForm {
+            hSection {
+                VStack(spacing: .padding4) {
+                    ForEach(vm.tiers, id: \.self) { tier in
+                        hRadioField(
+                            id: tier.name,
+                            leftView: {
+                                VStack(alignment: .leading, spacing: .padding8) {
+                                    HStack {
+                                        hText(tier.name)
+                                        Spacer()
+                                        hText(tier.premium.formattedAmountPerMonth)
+                                    }
+                                    hText(String(tier.productVariant.displayNameTierLong ?? ""))
+                                        .foregroundColor(hTextColor.Opaque.secondary)
+                                        .fixedSize()
+                                }
+                                .asAnyView
+                            },
+                            selected: $selectedTier,
+                            error: nil,
+                            useAnimation: true
+                        )
+                        .hFieldLeftAttachedView
+                    }
+                }
+            }
+            .sectionContainerStyle(.transparent)
+        }
+        .hFormAttachToBottom {
+            hSection {
+                VStack(spacing: .padding8) {
+                    hButton.LargeButton(type: .primary) {
+                        vm.setTier(for: self.selectedTier ?? "")
+                        selectTierNavigationVm.isEditTierPresented = false
+                    } content: {
+                        hText(L10n.generalContinueButton)
+                    }
+
+                    hButton.LargeButton(type: .ghost) {
+                        selectTierNavigationVm.isEditTierPresented = false
+                    } content: {
+                        hText(L10n.generalCancelButton)
+                    }
+
+                }
+            }
+            .sectionContainerStyle(.transparent)
+            .padding(.top, 16)
+        }
+        .onAppear {
+            self.selectedTier = vm.selectedTier?.name ?? vm.tiers.first?.name
+        }
+    }
+}
+
+#Preview{
+    EditTier(vm: .init())
+}
+
+extension Sequence {
+    func uniqued<Type: Hashable>(by keyPath: KeyPath<Element, Type>) -> [Element] {
```

Comment:
Do we use this function 'uniqued' somewhere?

Reply:
Nope! Just forgot to remove it!